src/psyclone/psyir/transformations/scalarization_trans.py

```diff
@@ -0,0 +1,324 @@
+
+    >>> from psyclone.psyir.backend.fortran import FortranWriter
+    >>> from psyclone.psyir.frontend.fortran import FortranReader
+    >>> from psyclone.psyir.transformations import ScalarizationTrans
+    >>> from psyclone.psyir.nodes import Loop
+    >>> code = """program test
+    ... integer :: i,j
+    ... real :: a(100), b(100)
+    ... do i = 1,100
+    ...   a(i) = i
+    ...   b(i) = a(i) * a(i)
+    ... end do
+    ... do j = 1, 100
+    ...  if(b(i) > 200) then
+    ...    print *, b(i)
+    ...  end if
+    ... end do
+    ... end program"""
+    >>> psyir = FortranReader().psyir_from_source(code)
+    >>> scalarise = ScalarizationTrans()
+    >>> scalarise.apply(psyir.walk(Loop)[0])
+    >>> print(FortranWriter()(psyir))
+    program test
+      integer :: i
+      integer :: j
+      real, dimension(100) :: a
+      real, dimension(100) :: b
+      real :: a_scalar
+      <BLANKLINE>
+      do i = 1, 100, 1
+        a_scalar = i
+        b(i) = a_scalar * a_scalar
+      enddo
+      do j = 1, 100, 1
+        if (b(i) > 200) then
+          ! PSyclone CodeBlock (unsupported code) reason:
+          !  - Unsupported statement: Print_Stmt
+          PRINT *, b(i)
+        end if
+      enddo
+    <BLANKLINE>
+    end program test
+    '''
+
+    @staticmethod
+    def _is_local_array(signature, var_accesses):
+        '''
+        :param signature: The signature to check if it is a local array symbol
+                          or not.
+        :type signature: :py:class:`psyclone.core.Signature`
+        :param var_accesses: The VariableAccessesInfo object containing
+                             signature.
+        :type var_accesses: :py:class:`psyclone.core.VariablesAccessInfo`
+        :returns bool: whether the symbol corresponding to signature is a
+                       local symbol or not.
+        '''
+        if not var_accesses[signature].is_array():
+            return False
+        base_symbol = var_accesses[signature].all_accesses[0].node.symbol
+        if not base_symbol.is_automatic:
+            return False
+
+        return True
+
+    @staticmethod
+    def _have_same_unmodified_index(signature, var_accesses):
+        '''
+        :param signature: The signature to check.
+        :type signature: :py:class:`psyclone.core.Signature`
+        :param var_accesses: The VariableAccessesInfo object containing
+                             signature.
+        :type var_accesses: :py:class:`psyclone.core.VariablesAccessInfo`
+        :returns bool: whether all the array accesses to signature use the
+                       same index, and whether the index is unmodified in
+                       the code region.
+        '''
+        array_indices = None
+        scalarizable = True
+        for access in var_accesses[signature].all_accesses:
+            if array_indices is None:
+                array_indices = access.component_indices
+            # For some reason using == on the component_lists doesn't work
```

Comment:
Is this comment outdated, we recently improved this, also the `[:]` shouldn't be needed (or is this what the comment is about?)

Reply:
The comment is about the [:] - when I removed it this didn't work

Reply:
I tested it, if I don't have the `[:]` it still fails

Reply:
Ok, this is strange but not for this PR, thanks for checking